### PR TITLE
Support for JAX-RS

### PR DESCRIPTION
**Pull request by muelletr**
Added public default constructors and some annotations to enable deserialization from JSON to Java.

### CODE REVIEW

**Comment by vruusmann** [comment]
Can you elaborate your use case? What is the motivation behind exporting BibTeX data structures into JSON (instead of BibTeX itself)?

There's also a BibJSON data format: http://okfnlabs.org/bibjson/ Perhaps it meets your use case better than plain old BibTeX data format?

File: pom.xml [hunk 1]
```diff
@@ -54,6 +54,11 @@
 			<version>4.12</version>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
```
Comment (koppor) [inline pom.xml]: I think, this has to be changed to the following:

```
<dependency>
  <groupId>com.fasterxml.jackson.core</groupId>
  <artifactId>jackson-annotations</artifactId>
  <version>2.9.2</version>
</dependency>
```

**Comment by koppor** [comment]
@muelletr develops a system for collaboratively improving BibTeX entries at https://github.com/muelletr/master-thesis-jaxrs, which will be the basis for [CloudRef](https://github.com/JabRef/cloudref). When using JBibTeX directly, the Angular app can directly read and write the BibTeX data.

I would directly usie JBibTeX, because:

1. BibTeX is the data format handled by the application (similar as [JabRef](https://www.jabref.org/))
2. The user should enter BibTeX and not some other format.
3. The conversion BibTeX <-> BibJSON currently seems to work in one direction only.

I agree that in future versions of CloudRef should be more sophisticated. Then, maybe both BibJSON and BibTeX needs to be supported. Nevertheless, with our current resources, I would like to "just" send BibTeX as JSON to an Angular application.

**Comment by vruusmann** [comment]
This issue appears to capture two sub-issues:
1. All JBibTeX "class model" classes should declare the default constructor (ie. a public no-arg constructor).
2. Annotate the `Value` class (and its selected subclasses) with some JSON-specific annotation classes. 

I don't like the idea of adding some Jackson dependency to an otherwise self-contained project. Maybe these Jackson annotation classes are available in some other dependency, or can be replaced with "standard" JAX_RS annotations? Will have to check.

**Comment by koppor** [comment]
(Just for commenting the issue - no solution currently)

Following annotations are used:

- `JsonIgnoreProperties(ignoreUnknown = true)` - [JSonIgnoreProperties](https://fasterxml.github.io/jackson-annotations/javadoc/2.6/com/fasterxml/jackson/annotation/JsonIgnoreProperties.html)
- `JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "type")` - [JsonTypeInfo](https://fasterxml.github.io/jackson-annotations/javadoc/2.4/com/fasterxml/jackson/annotation/JsonTypeInfo.html)

The Jackson module to interpret JAX-B annotations is the following: https://github.com/FasterXML/jackson-module-jaxb-annotations

**Comment by vruusmann** [comment]
On a second thought, a Jackson dependency could be declared with `provided` scope. This way it would not be included into "normal" builds, which expect to see a small and self-contained JBibTeX library JAR file. However, if there is a need to do some real (J)BibTeX-to-JSON binding stuff, then you would re-declare the Jackson dependency in your project's build configuration file, and have everything working as intended.

Still need to add some JUnit tests to make sure that the conversion between JBibTeX class model and JSON data format is taking place correctly.

**Comment by koppor** [comment]
@vruusmann We added the `provided` dependency and also added small JUnit tests. Do they meet your quality requirements or should be implement them in other ways?

**Comment by vruusmann** [comment]
@koppor Do you really need this BibTeX-to-JSON (and back) functionality integrated into the JBibTeX library? If you do, then I can make this happen.

Haven't been doing much research/development in this area for quite some time.

**Comment by koppor** [comment]
@vruusmann Yes, we need this for [CloudRef](https://github.com/JabRef/cloudref/), which offers collaborative management of bibliographies in the Cloud. We are short before an initial 1.0.0 release. Currently, we ship the JAR file with our modifications. I think, it would be good if that was really included in the library. On the one hand, I am aware that there are better ways such as [Data Transfer Objects](https://martinfowler.com/bliki/LocalDTO.html). On the other hand, we send and receive BibTeX as data object (AS IS!) and I don't see why we should reinvent a JSON format for BibTeX if JBibTex can be easily extended for that. [BibJSON](http://okfnlabs.org/bibjson/ is not an alternative, because its serialization differs from the one of a direct translation of BibTeX.

**Comment by vruusmann** [comment]
@koppor I've pushed JBibTeX version 1.0.16 to the Maven Central repository. Should become visible to the general public in a couple of hours time.

**Pull request closed without merging**